### DISCUSSION TITLE
How to get line  Segments array

### DISCUSSION THREAD

**Discussion by tomas-yang-fan** [iTwin/itwinjs-core#2876] [Q&A] [answered]
How to get line Segments array
![image](https://user-images.githubusercontent.com/56373600/145380991-9402e38f-372d-4855-bc97-50274e080c9b.png)

**Comment by pmconne** [comment, marked as answer]
The "Segments" array property is not actually stored as an array of segments - neither in the DGN nor the iModel. That is just how MicroStation's UI presents the underlying geometry.
In an iModel every geometric element has a [GeometryStream](https://www.itwinjs.org/learning/common/geometrystream/) property that contains its geometry.
You can iterate an element's geometry using a [GeometryStreamIterator](https://www.itwinjs.org/reference/imodeljs-common/geometry/geometrystreamiterator/).
While iterating you would need to inspect each entry in the stream to determine if it is a line string, and produce your segments array from the list of points defining the line string.

**Reply by tomas-yang-fan** [reply]
got it , thank you for you help
